kedro/extras/datasets/spark/spark_dataset.py

```diff
@@ -304,8 +325,41 @@ def __init__(  # pylint: disable=too-many-arguments
         if save_args is not None:
             self._save_args.update(save_args)
 
+        # Handle schema load argument
+        self._schema = self._load_args.pop("schema", None)
+        if self._schema is not None:
+            if isinstance(self._schema, dict):
+                self._schema = self._load_schema_from_file(self._schema)
+
         self._file_format = file_format
         self._fs_prefix = fs_prefix
+        self._handle_delta_format()
+
+    def _load_schema_from_file(self, schema: Dict[str, Any]) -> StructType:
+
+        filepath = schema.get("filepath")
+        if not filepath:
+            raise DataSetError(
+                "Schema load argument does not specify a `filepath` attribute. Please"
+                "include a path to a JSON serialized `pyspark.sql.types.StructType`."
+            )
+
+        credentials = deepcopy(schema.get("credentials")) or {}
+        protocol, schema_path = get_protocol_and_path(filepath)
+        file_system = fsspec.filesystem(protocol, **credentials)
+        pure_posix_path = PurePosixPath(schema_path)
+        load_path = get_filepath_str(pure_posix_path, protocol)
+
+        # Open schema file
+        with file_system.open(load_path) as fs_file:
+
+            try:
+                return StructType.fromJson(json.loads(fs_file.read()))
+            except Exception as exc:
+                raise DataSetError(
+                    f"Contents of `schema.filepath` ({schema_path}) are invalid. Please"
+                    f"provide a valid JSON serialized `pyspark.sql.types.StructType`."
+                ) from exc
 
         self._handle_delta_format()
```

Comment:
@MerelTheisenQB AFAIK this line was not added by my PR. I remember tackling a merge conflict here, could it be that this was due to an errornous merge?

Reply:
Fixed it, was a merge conflict issue! 💪🏻